coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -61,43 +61,81 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
       localPartitionPlanner.materialize(logicalPlan, qContext)
     }  else if (LogicalPlan.hasSubqueryWithWindowing(logicalPlan) || logicalPlan.isInstanceOf[TopLevelSubquery]) {
       materializeSubquery(logicalPlan, qContext)
-    } else {
-        walkLogicalPlanTree(logicalPlan, qContext).plans.head
+    } else logicalPlan match {
```

Comment:
Can you put the overall strategy as pseudocode in the javadoc of materialize method ? Could do for other planners as well - thank you for helping the next person who reads the planner code.

Reply:
Ah yes, I forgot that.. will do..